tmt/steps/execute/internal.py

```diff
@@ -151,11 +153,31 @@ def execute(self, test: Test, guest: Guest,
         # Create data directory, prepare test environment
         environment = self._test_environment(test, extra_environment)
 
+        test_wrapper_filepath = os.path.join(workdir, TEST_WRAPPER_FILENAME)
+
         # Prepare the test command (use default options for shell tests)
         if test.framework == "shell":
-            command = f"{tmt.utils.SHELL_OPTIONS}; {test.test}"
+            test_command = f"{tmt.utils.SHELL_OPTIONS}; {test.test}"
+        else:
+            test_command = test.test
+
+        self.write(test_wrapper_filepath, test_command, 'w')
+
+        os.chmod(test_wrapper_filepath, 0o755)
+
+        guest.push(
+            source=test_wrapper_filepath,
+            destination=test_wrapper_filepath,
+            options=["-p", "--chmod=755"]
+            )
+
+        remote_command = f'./{TEST_WRAPPER_FILENAME}'
+
+        if self.get('interactive'):
+            pass
+
         else:
-            command = test.test
+            remote_command = f'{tmt.utils.SHELL_OPTIONS}; {remote_command} </dev/null |& cat'
```

Comment:
The `SHELL_OPTIONS` are already in the wrapper. Why are we adding them here again?

Reply:
I'd say it's partially because I didn't realize it sooner to make the distinction more prominent. As I understand it, the options here are semantically different than `SHELL_OPTIONS`: `SHELL_OPTIONS` seem to be "options for running shell scripts", which is good, but here we need "options for running test wrapper". They do look the same, but that might be just a happy accident: one day developers may decide `set -eo pipefail` should be removed from `SHELL_OPTIONS`, but we absolutely need it to stay *here* and on this layer of execution (`SHELL_OPTIONS` apply to whatever is going on inside the wrapper).

How about this change? It should make the distinction more visible, not using the default shell options for running the wrapper itself.

```diff
diff --git a/tmt/steps/execute/internal.py b/tmt/steps/execute/internal.py
index 0d3d08eb..3eb0f40a 100644
--- a/tmt/steps/execute/internal.py
+++ b/tmt/steps/execute/internal.py
@@ -20,7 +20,8 @@ from tmt.steps.provision import Guest
 from tmt.utils import EnvironmentType
 
 TEST_WRAPPER_FILENAME = 'tmt-test-wrapper.sh'
-
+TEST_WRAPPER_INTERACTIVE_TRAMPOLINE = '{remote_command}'
+TEST_WRAPPER_NONINTERACTIVE_TRAMPOLINE = 'set -eo pipefail; {remote_command} </dev/null |& cat'
 
 @dataclasses.dataclass
 class ExecuteInternalData(tmt.steps.execute.ExecuteStepData):
@@ -172,10 +173,10 @@ class ExecuteInternal(tmt.steps.execute.ExecutePlugin):
         remote_command = f'./{TEST_WRAPPER_FILENAME}'
 
         if self.get('interactive'):
-            pass
+            remote_command = TEST_WRAPPER_INTERACTIVE_TRAMPOLINE.format(remote_command=remote_command)
 
         else:
-            remote_command = f'{tmt.utils.SHELL_OPTIONS}; {remote_command} </dev/null |& cat'
+            remote_command = TEST_WRAPPER_NONINTERACTIVE_TRAMPOLINE.format(remote_command=remote_command)
 
         # Prepare custom function to log output in verbose mode
         def log(
```

Reply:
I went ahead & pushed the change I proposed above in 45e9bd5c654d8c31645fc2e630f2d4ebcfdbd28d.